raiden/tests/smart_contracts/test_channel_manager.py

```diff
@@ -222,33 +226,100 @@ def test_channelmanager(private_keys, settle_timeout, tester_channelmanager):
-    nettingchannel.settle(sender=pkey0)
+    channels = list()
+    old_channel_addresses = list()
+    for partner_pkey in private_keys[1:]:
+        nettingcontract = new_nettingcontract(
+            pkey, partner_pkey, tester_state, log.append, tester_channelmanager, settle_timeout,
+        )
+        channels.append(nettingcontract)
 
-    tester_state.mine(1)
-    assert tester_state.block.get_code(nettingchannel.address) != 0
+        address = privatekey_to_address(partner_pkey)
+        channel_address = tester_channelmanager.getChannelWith(
+            address,
+            sender=pkey,
+        )
+        old_channel_addresses.append(channel_address)
+
+    for nettingchannel in channels:
+        netting_channel_settled(
+            tester_state,
+            nettingchannel,
+            pkey,
+            settle_timeout,
+        )
+
+    channels = list()
+    for partner_pkey in private_keys[1:]:
+        nettingcontract = new_nettingcontract(
+            pkey, partner_pkey, tester_state, log.append, tester_channelmanager, settle_timeout,
+        )
+        channels.append(nettingcontract)
+
+    # there must be a single entry for each participant
+    for partner_pkey in private_keys[1:]:
+        address = privatekey_to_address(partner_pkey)
+        channel_address = tester_channelmanager.getChannelWith(
+            address,
+            sender=pkey,
+        )
+        assert channel_address
+        assert channel_address not in old_channel_addresses
+
+
+@pytest.mark.parametrize('number_of_nodes', [5])
+def test_reopen_regression_bad_index_update(
+        private_keys,
+        settle_timeout,
+        tester_channelmanager,
+        tester_state):
+    """ deleteChannel used the wrong address to update the node_index mapping.
+        Correct usage would be
+
+        (addr0, addr1) => channel_idx
+
+    But instead of overwriting the existing index, a new entry was added as:
+
+        (addr0, channel_addr) => channel_idx
+    """
+    pkey0 = private_keys[0]
+    pkey1 = private_keys[1]
+    pkey2 = private_keys[2]
+    addr1 = privatekey_to_address(pkey1)
 
-    # Now that channel with address1 is settled a new one can be opened.
-    # Old entry will be deleted when calling newChannel
     tester_channelmanager.newChannel(
-        address1,
+        addr1,
         settle_timeout,
         sender=pkey0,
     )
 
-    # make sure the old channel is properly selfdestructed
-    tester_state.mine(1)
-    assert tester_state.block.get_code(nettingchannel.address) != 0
+    log = list()
+    nettingchannel = new_nettingcontract(
+        pkey0, pkey2, tester_state, log.append, tester_channelmanager, settle_timeout,
+    )
+
+    netting_channel_settled(
+        tester_state,
+        nettingchannel,
+        pkey0,
+        settle_timeout,
+    )
+
+    nettingchannel = new_nettingcontract(
+        pkey0, pkey2, tester_state, log.append, tester_channelmanager, settle_timeout,
+    )
```

Comment:
It's a regression test, the older implementation failed with these steps on the last `newChannel`.

Even though `test_reopen_channel` and `test_reopen_regression_bad_index_update` do have some overlap, I left this as a minimal alternative, my idea is to expand the scenarios covered in `test_reopen_channel`.